### ISSUE TITLE
Encapsulate required data types in at_onboarding_flutter

### ISSUE THREAD

**Issue by ralfeus** [atsign-foundation/at_widgets#583]
### Which at_widget package you find issue with?

at_onboarding_flutter

### On what platforms did you experience the issue?

Android, iOS, Windows, MacOS, Linux

### What version are you using?

at_onboarding_flutter: 5.0.5

### What happened?

`AtOnboarding.onboard()` requires a `config` parameter of type `AtOnboardingConfig`, which is a part of `at_onboarding_flutter` package. Its constructor requires in its turn `atClientPreference` parameter of type `AtClientPreference`. This type is not a part of `at_onboarding_flutter` nor exposed by one. So it's hard to figure what else should be added to dependencies to make the project running.
I recommend to expose `AtClientPreference` by `at_onboarding_flutter` package

### Steps to reproduce

```bash
1. Create widget from example:

  Widget build(BuildContext context) {
      final atClientPreference = AtClientPreference();
      return TextButton(
      onPressed: () async {
        final result = await AtOnboarding.onboard(
          context: context,
          config: AtOnboardingConfig(
            atClientPreference: atClientPreference,
            domain: 'atkrypto.io',
            rootEnvironment: RootEnvironment.Testing,
            appAPIKey: AtEnv.appApiKey,
          ),
        );
    }

2. Observe impossibility to compile
...
```

**Comment by ksanty** [comment]
Is this a documentation update @sachins-geekyants

**Comment by sonle-geekyants** [comment]
https://github.com/atsign-foundation/at_widgets/pull/594 This is my PR, please check it @sachins-geekyants @ralfeus

**Comment by sachins-geekyants** [comment]
Fix released in `at_onboarding_flutter: ^6.0.1`
